compiler/rustc_lint/src/macro_expr_fragment_specifier_2024_migration.rs

```diff
@@ -12,14 +12,38 @@ use rustc_session::lint::FutureIncompatibilityReason;
 use rustc_span::edition::Edition;
 use rustc_span::sym;
 
-use crate::lints::MacroExprFragment2023;
+use crate::lints::MacroExprFragment2024;
 use crate::EarlyLintPass;
 
 declare_lint! {
+    /// TODO: better explaination for this.
```

Comment:
I am not good at writing documentation, @traviscross  or @eholk have some suggestions on what to write here?